src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length < 1) {
+      return '';
+    }
+
+    const arr = [...this];
+
+    for (let i = 0; i < arr.length; i++) {
+      if (arr[i] === undefined || arr[i] === null) {
+        arr[i] = '';
+      }
+    }
+
+    let str = `${arr[0]}`;
+
+    for (let i = 1; i < arr.length; i++) {
+      str += `${separator}${arr[i]}`;
+    }
```

Comment:
you can minimize it to one loop. if add arr[i] or '', then separator. If last element - don't add separator